lib/src/entrypoint.dart

```diff
@@ -720,9 +724,22 @@ class Entrypoint {
     }
 
     final packagePathsMapping = <String, String>{};
-    for (final pkg in cfg.packages) {
-      // Pub always sets packageUri = lib/
-      if (pkg.packageUri == null || pkg.packageUri.toString() != 'lib/') {
+
+    // We allow the package called 'flutter_gen' to be injected into
+    // package_config.
+    //
+    // This is somewhat a hack. But it allows flutter to generate code in a
+    // package as it likes.
+    //
+    // See https://github.com/flutter/flutter/issues/73870 .
+    final packagesToCheck =
+        cfg.packages.where((package) => package.name != 'flutter_gen');
+    for (final pkg in packagesToCheck) {
+      // Pub always sets packageUri = 'lib/' but package:package_config might
+      // change it to 'lib'.
+      if (pkg.packageUri == null ||
+          (pkg.packageUri.toString() != 'lib' &&
+              pkg.packageUri.toString() != 'lib/')) {
```

Comment:
We could patch our `package_config.json` parser to append `/` when it is omitted, given that spec says to do so.

Reply:
Yeah - that's much better!

Done